evalml/automl/automl_search.py

```diff
@@ -126,16 +128,19 @@ def __init__(self,
                 to `multiclass` or `regression` depending on the problem type. Note that if allowed_pipelines is provided,
                 this parameter will be ignored.
 
-            data_split (sklearn.model_selection.BaseCrossValidator): data splitting method to use. Defaults to StratifiedKFold.
+            data_split (sklearn.model_selection.BaseCrossValidator): Data splitting method to use. Defaults to StratifiedKFold.
 
-            tuner_class: the tuner class to use. Defaults to scikit-optimize tuner
+            tuner_class: The tuner class to use. Defaults to SKOptTuner.
 
-            start_iteration_callback (callable): function called before each pipeline training iteration.
-                Passed three parameters: pipeline_class, parameters, and the AutoMLSearch object.
+            start_iteration_callback (callable): Function called before each pipeline training iteration.
+                Passed three parameters: The pipeline class, the pipeline parameters, and the AutoMLSearch object.
 
-            add_result_callback (callable): function called after each pipeline training iteration.
+            add_result_callback (callable): Function called after each pipeline training iteration.
                 Passed three parameters: A dictionary containing the training results for the new pipeline, an untrained_pipeline containing the parameters used during training, and the AutoMLSearch object.
 
+            error_callback (callable): Function called when `search()` errors and raises an Exception.
+                Passed two positional parameters: The Exception raised and the AutoMLSearch object. Accepts kwargs.
```

Comment:
nit-pick: I think we should say "must accept kwargs"? Right now fold_num and pipeline are always passed as kwargs.

Reply:
Agreed about "must accept kwargs" and @jeremyliweishih suggestion. Would be helpful to describe what the kwargs part is for, how to use it. Will respond to `Protocol` discussion elsewhere.